fem/coefficient.hpp

```diff
@@ -1858,6 +1858,85 @@ class TraceCoefficient : public Coefficient
                const IntegrationPoint &ip) override;
 };
 
+/// Scalar coefficient defined as component of a vector coefficient
+class VectorComponentCoefficient : public Coefficient
+{
+private:
+   VectorCoefficient *a;// = nullptr;
+
+   mutable Vector va;
+   int i;
```

Comment:
A longer, more descriptive name like `component` may be safer.

Reply:
Changed